Source/Scene/Scene.js

```diff
@@ -3645,6 +3664,85 @@ define([
         camera.right = right;
     }
 
+    function updateAsyncLoader(scene, asyncLoader) {
+        var context = scene._context;
+        var uniformState = context.uniformState;
+        var frameState = scene._frameState;
+
+        var view = scene._pickOffscreenView;
+        scene._view = view;
+
+        var ray = asyncLoader.ray;
+        var primitives = asyncLoader.primitives;
+
+        updateCameraFromRay(ray, view.camera);
+
+        updateFrameState(scene);
+        frameState.passes.offscreen = true;
+        frameState.passes.async = true;
+
+        uniformState.update(frameState);
+
+        var commandList = frameState.commandList;
+        var commandsLength = commandList.length;
+
+        var ready = true;
+        var primitivesLength = primitives.length;
+        for (var i = 0; i < primitivesLength; ++i) {
+            var primitive = primitives[i];
+            if (scene.primitives.contains(primitive) && primitive.show) {
+                // Only update primitives that are still contained in the scene's primitive collection and are still visible
+                ready = primitive.update(frameState) && ready;
+            }
+        }
+
+        // Ignore commands pushed during async pass
+        commandList.length = commandsLength;
+
+        scene._view = scene._defaultView;
+
+        if (ready) {
+            frameState.afterRender.push(function() {
+                asyncLoader.deferred.resolve();
+            });
+        }
+
+        return ready;
+    }
+
+    function updateAsyncLoaders(scene) {
+        var asyncLoaders = scene._asyncLoaders;
+        for (var i = 0; i < asyncLoaders.length; ++i) {
```

Comment:
Are we still storing `length` before a `for` loop?  If not, are we sure?  And update the Coding Guide if it is still in there.

Reply:
This is intentional because the array size can get modified during the iteration. Like when an async loader finishes. I can leave a comment for that.